### PR TITLE
Fix and Enable Clang compilation

### PR DESCRIPTION
**Pull request by dbrageul**

### CODE REVIEW

File: parameter/Android.mk [hunk 1]
```diff
@@ -186,7 +185,7 @@ LOCAL_STATIC_LIBRARIES := libxmlserializer_host libpfw_utility_host libxml2
 
 LOCAL_LDLIBS += -ldl
 
-LOCAL_CLANG := false
+#LOCAL_CLANG := false
```
Comment (krocard) [inline parameter/Android.mk]: dead code

**Comment by clero** [comment]
travis compilation is fixed in #94 (just some comments which need to be fixed)

File: bindings/python/CMakeLists.txt [hunk 2]
```diff
@@ -54,13 +54,11 @@ swig_link_libraries(PyPfw parameter ${PYTHON_LIBRARIES})
...
 set_property(TARGET _PyPfw PROPERTY LIBRARY_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 
-# The 'unused-but-set-variable' warning must be disabled because SWIG generates
-# files that do not respect that contraint.
 # '-DSWIG_PYTHON_SILENT_MEMLEAK' is needed because the "memleak" warning
 # pollutes the standard output. At the time of writing, the only warning is
 # spurious anyway, as it relates to "ILogger *" which is an abstract
 # class/interface class and as such cannot be destroyed.
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-but-set-variable -DSWIG_PYTHON_SILENT_MEMLEAK")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DSWIG_PYTHON_SILENT_MEMLEAK")
```
Comment (krocard) [inline bindings/python/CMakeLists.txt]: Maybe add -Wno-error and add a comment explaining that the code is generated by swig and thus can not be fixed even if warned.

**Comment by krocard** [comment]
Uncomment # clang in [.travis.yml:12](https://github.com/01org/parameter-framework/blob/master/.travis.yml#L12)

**Comment by dawagner** [comment]
All these commits have been merged in #133.

**Pull request closed without merging**